code/messaging/src/main/java/com/adobe/marketing/mobile/PropositionItem.java

```diff
@@ -124,6 +135,70 @@ Inbound decodeContent() {
         return inboundContent;
     }
 
+    /**
+     * Creates a {@code PropositionItem} object from the provided {@code Map<String, Object>}.
+     *
+     * @param eventData {@link Map<String, Object>} event data
+     * @return {@link PropositionItem} object created from the provided {@code Map<String, Object>}.
+     */
+    static PropositionItem fromEventData(final Map<String, Object> eventData) {
+        PropositionItem propositionItem = null;
+        try {
+            final String uniqueId = DataReader.optString(eventData, PAYLOAD_ID, "");
+            final String schema = DataReader.optString(eventData, PAYLOAD_SCHEMA, "");
+            final Map<String, Object> dataMap = DataReader.optTypedMap(Object.class, eventData, PAYLOAD_DATA, null);
+
+            final StringBuilder contentStringBuilder = new StringBuilder();
+            if (schema.equals(JSON_CONTENT_SCHEMA_VALUE)) {
+                final JSONObject contentJSON = new JSONObject(DataReader.optTypedMap(Object.class, dataMap, PAYLOAD_CONTENT, null));
+                final JSONArray rulesList = contentJSON.getJSONArray(DATA_RULES_KEY);
```

Comment:
Rules might not always be present (e.g. code based JSON offers), so let's not try to parse them here.